### PR TITLE
PLT-74: Enable Up Arrow keyboard shortcut to edit your last message

### PR DESCRIPTION
**Pull request by florianorben**
Hi,

as described in PLT-74 this change enables you to edit your last post by pressing the "up arrow" while focused in the create_post input box.
The way it works now is, that you can always edit your last post, as long as the input is empty.
Changing it so that you could only edit your last post within the next ten (or whatever) seconds, would be fairly easy to implement - just drop a message if that should be done.

Cheers, Flo

### CODE REVIEW

File: web/react/components/edit_post_modal.jsx [hunk 1]
```diff
@@ -59,6 +61,18 @@ export default class EditPostModal extends React.Component {
     handleUserInput(e) {
         this.setState({editText: e.target.value});
     }
+    handleEditPostEvent(options) {
+        this.setState({
+            editText: options.message || '',
+            title: options.title || '',
+            post_id: options.postId || '',
```
Comment (jwilander) [inline web/react/components/edit_post_modal.jsx]: I think this should be `post_id: options.lastPostId` correct?

**Comment by jwilander** [comment]
Good stuff @florianorben! Other than my one comment the code looks good

File: web/react/components/create_post.jsx [hunk 2]
```diff
@@ -292,6 +294,24 @@ export default class CreatePost extends React.Component {
...
+        if (e.keyCode === KeyCodes.UP && this.state.messageText === '') {
+            e.preventDefault();
+
+            const channelId = ChannelStore.getCurrentId();
+            const lastPost = PostStore.getCurrentUsersLatestPost(channelId);
+            var type = (lastPost.root_id && lastPost.root_id.length > 0) ? 'Comment' : 'Post';
+
+            AppDispatcher.handleViewAction({
+                type: ActionTypes.RECIEVED_EDIT_POST,
+                refoucsId: '#post_textbox',
```
Comment (hmhealey) [inline web/react/components/create_post.jsx]: Small typo here

File: web/react/components/edit_post_modal.jsx (hunk 1 above)
Comment (hmhealey) [inline web/react/components/edit_post_modal.jsx]: Yeah the post_id isn't being set currently, but that explains why I'm getting a server error when testing this

**Comment by hmhealey** [comment]
A couple other things that I noticed while testing:
1) If you hit the up arrow in a channel where you haven't posted before, it throws a TypeError on like 303 of create_post.jsx because it can't find your last post. It just needs a bit of error handling for that.
2) Are you also able to make it focus the textbox in the edit post dialog when it is displayed? I'm surprised we don't do this already with that dialog, but that would would make it so that you could edit your last post without needing to use the mouse at all which would be awesome.

File: web/react/components/edit_post_modal.jsx (hunk 1 above)
Comment (florianorben) [inline web/react/components/edit_post_modal.jsx]: Yep should be lastPostId, don't know why it worked for me probably some caching in there, I'm going to fix it later this eve

File: web/react/components/create_post.jsx (hunk 2 above)
Comment (florianorben) [inline web/react/components/create_post.jsx]: I guess you mean ````RECIEVED_EDIT_POST````?
"RECEIVED" has this typo in all jsx files, I used this spelling to be in line with the other event names.
Should I fix this and/or the other typos or submit an additional merge request?

Comment (jwilander) [inline web/react/components/create_post.jsx]: Actually I think he meant `refoucsId` should be `refocusId`

**Comment by florianorben** [comment]
> 1) If you hit the up arrow in a channel where you haven't posted before, it throws a TypeError on like 303 of create_post.jsx because it can't find your last post. It just needs a bit of error handling for that.

Good catch, haven't thought of that


> 2) Are you also able to make it focus the textbox in the edit post dialog when it is displayed? I'm surprised we don't do this already with that dialog, but that would would make it so that you could edit your last post without needing to use the mouse at all which would be awesome.

Yep, will do. - I'll also add that for the "edit" link in the post info's dropdown. 


I'm going submit my changes later this evening - need some time off first ;)

**Comment by florianorben** [comment]
Hey guys,

- bugs are fixed now
- renamed lastPostId to postId (so that the open-edit-event _could_ also be triggered from somewhere else, without the lastPostEdit sounding weird in that context)
- typo fixed ;) didnt really see that - point still stands, should recieve be renamed to receive?
- textarea will now automatically be focused

**Comment by hmhealey** [comment]
Awesome, works great. Thanks for fixing those things.

And recieve probably shouldn't be fixed right now since it's misspelled that way all over the clientside code. We might go in and fix that everywhere at some point, but it's best to keep it consistent for now.

**Comment by florianorben** [comment]
> Awesome, works great. Thanks for fixing those things.

> And recieve probably shouldn't be fixed right now since it's misspelled that way all over the clientside code. We might go in and fix that everywhere at some point, but it's best to keep it consistent for now.

ok ;)

**Pull request merged**